src/rapids_singlecell/preprocessing/_utils.py

```diff
@@ -129,26 +120,19 @@ def __mean_var(X_part, minor, major):
         )
         return mean, var
 
-    parts = client.sync(_extract_partitions, X)
-    futures = [
-        client.submit(__mean_var, part, minor, major, workers=[w]) for w, part in parts
+    blocks = X.to_delayed().ravel()
+    mean_var_blocks = [
+        da.from_delayed(
+            __mean_var(block, minor, major),
+            shape=(2, X.chunks[0][ind]),
+            dtype=X.dtype,
+            meta=cp.array([]),
+        )
+        for ind, block in enumerate(blocks)
     ]
```

Comment:
Nothing in the public API tests this, and it's generally not great practice to test internal functions.  I'll have to have a closer look.